metron-stellar/stellar-common/pom.xml

```diff
@@ -230,11 +230,29 @@
             <version>${global_httpclient_version}</version>
         </dependency>
         <dependency>
-            <!-- junit dependency added with default scope to allow inclusion of StellarProcessorUtils in main jar.
-                 It is excluded from the uber-jar. -->
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>${global_junit_version}</version>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-core</artifactId>
+            <version>${global_hamcrest_version}</version>
+            <scope>test</scope>
+        </dependency>
+        <!-- junit dependency added with default scope to allow inclusion of StellarProcessorUtils in main jar.
+             It is excluded from the uber-jar. -->
+        <dependency>
```

Comment:
@nickwallen Why are we using JUnit assertions in `StellarProcessorUtils`, rather than either throwing something else (and an `AssertionError` from `fail` is an`Error` anyway under the hood if these checks fail)? It seems really odd that we're using JUnit utilities to enforce correctness of `src`.

If we were to migrate to something else(and it doesn't have to be in this PR), it'd mean this dep gets dropped back down to test, and cleans up the uber jar definition. I just want to make sure I'm not misunderstanding something here.

Reply:
While digging around for all the JUnit that makes its way into our uber jars, I tried just doing the checks and throwing `AssertionError` and it seems to work just fine.

https://github.com/justinleet/metron/blob/junit5_metron-2310/metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/utils/StellarProcessorUtils.java

Should be easy enough to port it back to master, and possibly clean it up.

Reply:
I agree with you @justinleet .  It doesn't make sense to use JUnit assertions in non-test code like `ensureKryoSerializable` and `ensureJavaSerializable`.

Reply:
@nickwallen I'm going to resolve this conversation, since it got taken care of as part of https://github.com/apache/metron/pull/1561. Let me know if there's any other input you have or want to see.